### PR TITLE
Arduino version bump and required dependencies

### PR DESCRIPTION
**Pull request by mapmot**
Initial and very hackish attempt at getting the latest version of Arduino into portage. Mostly based on https://github.com/maurerpe/arduino-overlay, and https://bugs.gentoo.org/525882

*dev-util/ctags is an ugly hack to force the arduino version of ctags;
*jackson-modules-base is a fast workaround to get jackson-modules-mrbean, needed for the Arduino IDE
*the dev-go/* ebuilds are needed for arduino-builder

### CODE REVIEW

**Comment by chewi** [comment]
I appreciate the work you've done but we can't take this as one big commit. Please break it up and commit changes with repoman.

File: dev-embedded/arduino/arduino-1.8.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+			    --java_args "-DAPP_DIR=/usr/share/${PN} -DCORE_DIR=${CORE} -splash:/usr/share/${PN}/lib/splash.png"
+
+	if use doc; then
+		dodoc revisions.txt "${S}"/README.md
+		dohtml -r reference
+	fi
+
+	# Install menu and icons
+	domenu "${FILESDIR}/${PN}.desktop"
+	for sz in `ls lib/icons | sed -e 's/\([0-9]*\)x[0-9]*/\1/'`; do
```
Comment (jonasstein) [inline dev-embedded/arduino/arduino-1.8.5.ebuild]: ```local sz``` missing here

File: dev-embedded/arduino/arduino-1.8.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+JAVA_ANT_REWRITE_CLASSPATH="yes"
+
+S="${WORKDIR}/Arduino-${PV}"
+CORE="/usr/share/arduino"
+SHARE="/usr/share/${PN}"
+
+src_prepare() {
+	# Remove bundled libraries to ensure the system libraries are used
+	# Elegant, but breaks the build :(
+	#rm -f {arduino-core,app}/lib/{apple*,batik*,bcpg*,bcprov*,commons-[^e]*,jackson-*,jmdns*,jna*,jsch*,jssc*,xmlgraphics*} || die
```
Comment (jonasstein) [inline dev-embedded/arduino/arduino-1.8.5.ebuild]: remove obsolete lines

File: dev-embedded/arduino-builder/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>junk4me46806@yahoo.com</email>
+	</maintainer>
+</pkgmetadata>
```
Comment (jonasstein) [inline dev-embedded/arduino-builder/metadata.xml]: use ```metagen``` and fill out more fields https://devmanual.gentoo.org/ebuild-writing/misc-files/metadata/

**Comment by jonasstein** [comment]
I have added the do not merge label, because you should fix some more things and split it in small commits with ```repoman ci``` in a new PR.

**Pull request closed without merging**